flow/project.py

```diff
@@ -2482,21 +2487,23 @@ def _fetch_status(
 
         def compute_status(data):
             scheduler_id, scheduler_status, aggregate_id, aggregate, group = data
-            # Only single-operation groups are supported for status fetching
-            assert len(group.operations) == 1
 
             status = {}
             error_text = None
+            is_user_defined_group = group.name not in self._operations
             try:
                 status["scheduler_status"] = scheduler_status
                 completed = group._complete(aggregate)
                 status["completed"] = completed
                 # If the group is not completed, it is sufficient to determine
                 # eligibility while ignoring postconditions (we know at least
                 # one postcondition is not met).
-                status["eligible"] = not completed and group._eligible(
-                    aggregate, ignore_conditions=IgnoreConditions.POST
-                )
+                if is_user_defined_group:
+                    status["eligible"] = group._eligible(aggregate)
+                else:
+                    status["eligible"] = not completed and group._eligible(
+                        aggregate, ignore_conditions=IgnoreConditions.POST
+                    )
```

Comment:
Good catch @b-butler. It seems like this should use the condition `len(group.operations) > 1` instead of `is_user_defined_group = group.name not in self._operations`. A user-defined group with a single operation would still be able to use `not completed` as a shortcut for determining its eligibility, in my understanding. That would help avoid some extra calls to `group._eligible`.

Edge case I have spent less than ten seconds considering: can a group have zero operations? (Probably, by defining it and assigning no operations.) Is the eligibility/completion state of such a group defined in a consistent way for status checks and execution?

Reply:
The reason I used that check rather than your suggestion was to only check once since `is_user_defined_group` is used specifically in the user defined sense below, but we can do the length check here to utilize the short cut, but I am not sure if that will be much change in most cases as a single operation user defined group is likely rare.

Also, yes such operations would register right now as being complete and not eligible (according to the logic of the builtins `any` and `all`).